packages/blockchain-wallet-v4-frontend/src/scenes/plugin/Activity/SelectCoin/SelectCoin.tsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react'
+import { useIntl } from 'react-intl'
+import { Icon } from '@blockchain-com/constellation'
+import { IconArrowLeft } from '@blockchain-com/icons'
+import styled from 'styled-components'
+
+import { Flex } from 'components/Flex'
+import { Padding } from 'components/Padding'
+
+import { CoinDataItem } from '../../../../hooks/useSelfCustodyCoinsBalances'
+import SearchInput from './SearchInput'
+import SelectCoinItem from './SelectCoinItem'
+
+const Wrapper = styled(Flex)`
+  height: 100%;
+`
+
+const IconWrapper = styled(Flex)`
+  cursor: pointer;
+`
+
+const SearchInputWrapper = styled(Flex)`
+  margin-bottom: 27px;
+  width: 100%;
+`
+
+export interface ISearchCoinProps {
+  coins: CoinDataItem[]
+  onClose(): void
+  onSelectCoin(coin: string): void
+}
+
+const SelectCoin: React.FC<ISearchCoinProps> = ({ coins, onClose, onSelectCoin }) => {
+  const [searchInputValue, setSearchInputValue] = React.useState('')
+  const intl = useIntl()
+
+  const handleBackClick = () => {
+    onClose()
+  }
+
+  const handelSearchChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchInputValue(e.target.value)
+  }
+
+  const ethWithBalance = coins.find((item) => item.coinfig.symbol === 'ETH')
```

Comment:
for calculations like this we should probably use some memoization to avoid redundant rerenders/memory allocations. So in this case

```
const ethWithBalance = useMemo(() => {
  return coins.find((item) => item.coinfig.symbol === 'ETH')
}, [coins])
```

Reply:
As for the use of memoization (here and all the places that you mentioned below) - it's not always optimal. In my experience memoization is usually more expensive in places like this than it is to recalculate each time. I'll go through the code again quickly, run some benchmarks and add memoization where needed.